docs/proposals/additional_auth.md

```diff
@@ -0,0 +1,88 @@
+# Overview
+Currently the only authorization method supported by the service catalog is
+basic auth. The user is expected to create a secret containing their
+credentials which is then read by the service catalog when required.
+
+# Problem
+We would like to be able to support brokers which require other kinds of
+authorization. Hopefully we can add support in a way that is generic enough
+for new types of authorization to be easy to support.
+
+# Background
+Currently the way auth works is
+* parse the secret information and generate an `AuthConfig` (`getAuthCredentialsFromBroker`)
+* use said `AuthConfig` to generate a ClientConfig and subsequently a broker client
+* broker client uses the `AuthConfig` to do some auth stuff on each request if necessary (`prepareAndDo`)
+
+AuthConfig is currently a struct
+
+```
+type AuthConfig struct {
+	BasicAuthConfig *BasicAuthConfig
+}
+```
+The auth information is a struct inside the `BrokerSpec`
+
+```
+type BrokerAuthInfo struct {
+	BasicAuthSecret *v1.ObjectReference
+}
+```
+
+# Proposal
+We can add support for other types of auth by adding a new field to the
+BrokerAuthInfo struct which identifies what type of auth we want to use. For
+simplicity, we use a string here. We can just rename `BasicAuthSecret` to be
+`AuthSecret`
+
+```
+type BrokerAuthInfo struct {
+	AuthSecret *v1.ObjectReference
+	SecretType string // the supported values are "basic"
```

Comment:
> in case someone wanted to use that for parsing the secret

@rifung can you clarify please? there is already constant for basic auth, why don't we use the same approach for other auth types? https://github.com/kubernetes/api/blob/master/core/v1/types.go#L4329